src/m365/aad/commands/app/app-set.ts

```diff
@@ -206,13 +214,94 @@ class AadAppSetCommand extends GraphCommand {
       .then(_ => Promise.resolve(objectId));
   }
 
+  private configureCertificate(args: CommandArgs, objectId: string, logger: Logger): Promise<void> {
+    if (!args.options.certificateFile && !args.options.certificateBase64Encoded) {
+      return Promise.resolve();
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Setting certificate for Azure AD app...`);
+    }
+
+    return this.getCertificateBase64Encoded(args, logger).then((certificateBase64Encoded) => {
+      const getAppRequestOptions: AxiosRequestConfig = {
+        url: `${this.resource}/v1.0/myorganization/applications/${objectId}?$select=keyCredentials`,
+        headers: {
+          'content-type': 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      return request
+        .get<Application>(getAppRequestOptions)
+        .then((application: Application): Promise<void> => {
+          
+          const keyCredentials = application.keyCredentials as KeyCredential[];
+          
+          // The graph types of  defines the 'key' property of KeyCredential as 'NullableOption<number>'
+          // while it is a base64 encoded string. This is any is used here.
```

Comment:
This would be very good to report to the Graph team. Have you done it already by any chance?

Reply:
No, but I will 👍